src/qibo/backends/__init__.py

```diff
@@ -167,3 +167,10 @@ def set_threads(nthreads):
     if nthreads < 1:
         raise_error(ValueError, "Number of threads must be positive.")
     GlobalBackend().set_threads(nthreads)
+
+
+def _check_backend(backend):  # pragma: no cover
```

Comment:
I think this way it would cause some confusion with `set_backend`

Reply:
Ok, that's reasonable.

At a second thought, this mechanism is likely to be used more internally than by users, and since backends' internals are already complex enough, let's limit public usage to `set_backend()`.